drivers/storage/s3fs/storage/s3fs_storage.go

```diff
@@ -0,0 +1,392 @@
+}
+
+func newDriver() types.StorageDriver {
+	return &driver{}
+}
+
+func (d *driver) Name() string {
+	return s3fs.Name
+}
+
+// Init initializes the driver.
+func (d *driver) Init(ctx types.Context, config gofig.Config) error {
+	d.config = config
+
+	fields := log.Fields{}
+
+	var (
+		accessKey  string
+		secretKey  string
+		region     string
+		endpoint   string
+		maxRetries int
+	)
+
+	d.tag = d.config.GetString(s3fs.ConfigS3FSTag)
+	fields[s3fs.Tag] = d.tag
+
+	accessKey = d.config.GetString(s3fs.ConfigS3FSAccessKey)
+	fields[s3fs.AccessKey] = accessKey
+
+	secretKey = d.config.GetString(s3fs.ConfigS3FSSecretKey)
+	if secretKey != "" {
+		fields[s3fs.SecretKey] = "******"
+	}
+
+	region = d.config.GetString(s3fs.ConfigS3FSRegion)
+	fields[s3fs.Region] = region
+
+	endpoint = d.config.GetString(s3fs.ConfigS3FSEndpoint)
+	fields[s3fs.Endpoint] = endpoint
+
+	maxRetries = d.config.GetInt(s3fs.ConfigS3FSMaxRetries)
+	fields[s3fs.MaxRetries] = maxRetries
+
+	ctx.WithFields(fields).Debug("s3 service connection attempt")
+	sess := session.New()
+
+	var (
+		awsLogger   = &awsLogger{ctx: ctx}
+		awsLogLevel = aws.LogOff
+	)
+	if ll, ok := context.GetLogLevel(ctx); ok {
+		switch ll {
+		case log.DebugLevel:
+			awsLogger.lvl = log.DebugLevel
+			awsLogLevel = aws.LogDebugWithHTTPBody
+		case log.InfoLevel:
+			awsLogger.lvl = log.InfoLevel
+			awsLogLevel = aws.LogDebug
+		}
+	}
+
+	d.svc = awss3.New(sess, &aws.Config{
+		Region:     &region,
+		Endpoint:   &endpoint,
+		MaxRetries: &maxRetries,
+		Credentials: credentials.NewChainCredentials(
+			[]credentials.Provider{
+				&credentials.StaticProvider{
+					Value: credentials.Value{
+						AccessKeyID:     accessKey,
+						SecretAccessKey: secretKey,
+					},
+				},
+				&credentials.EnvProvider{},
+				&credentials.SharedCredentialsProvider{},
+				&ec2rolecreds.EC2RoleProvider{
+					Client: ec2metadata.New(sess),
+				},
+			},
+		),
+		Logger:   awsLogger,
+		LogLevel: aws.LogLevel(awsLogLevel),
+	})
+	ctx.WithFields(fields).Debug("s3 connection created")
+
+	if _, err := d.svc.ListBuckets(&awss3.ListBucketsInput{}); err != nil {
+		ctx.WithFields(fields).WithError(err).Error("s3 connection failed")
+		return err
+	}
+
+	ctx.WithFields(fields).Info("storage driver initialized")
+	return nil
+}
+
+type awsLogger struct {
+	lvl log.Level
+	ctx types.Context
+}
+
+func (a *awsLogger) Log(args ...interface{}) {
+	switch a.lvl {
+	case log.DebugLevel:
+		a.ctx.Debugln(args...)
+	case log.InfoLevel:
+		a.ctx.Infoln(args...)
+	}
+}
+
+func mustInstanceIDID(ctx types.Context) *string {
+	return &context.MustInstanceID(ctx).ID
+}
+
+// NextDeviceInfo returns the information about the driver's next available
+func (d *driver) NextDeviceInfo(
+	ctx types.Context) (*types.NextDeviceInfo, error) {
+	return s3fsUtils.NextDeviceInfo, nil
+}
+
+// Type returns the type of storage the driver provides.
+func (d *driver) Type(ctx types.Context) (types.StorageType, error) {
+	return types.Object, nil
```

Comment:
Is this true? Even though the backing storage is object, you end up with a file system, so I would think the type is `File`.